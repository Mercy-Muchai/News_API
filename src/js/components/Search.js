import React from 'react';
import ReactDOM from 'react-dom';
import axios from 'axios';

export default class Search extends React.Component {
  constructor() {
    super();
    this.state = {
      sources: [],
      search: 'Search Source'
    };
  }
  updateSearch(event) {
    this.setState({ search: event.target.value });

  }

  componentDidMount() {
    axios.get(`https://newsapi.org/v1/sources?language=en`)
      .then((result) => {
        this.setState({
          sources: result.data.sources
        });
      });
  }

  render() {
    let found = this.state.sources.filter(
      (sources) => {
        return sources.name.toLowerCase().indexOf(this.state.search.toLowerCase()) !== -1;

      });

    return (
      <div className="col-sm-12" classID="search" style={{ paddingTop: "25px" }}>

        <input type="text" placeholder='Search Source' value={this.state.search} onChange={this.updateSearch.bind(this)} style={{ backgroundColor: '#ffffff' }} className="searchbox" />
        <span id="tweet-container-index" style={{ paddingLeft: '1010px'}}>
          <a href="https://twitter.com/share" className="twitter-share-button" data-url={'https://getnews-api.herokuapp.com/#/abc-news-au/top'} data-text={'Get News here from all over the world'}>Tweet</a>
        </span>
        {found.map((sources) => {
          return (
            <div className="search" key={sources.id}>
              <a href={sources.url}> {sources.name} </a>
            </div>);
        }
        )}

      </div>
    );
  }
};