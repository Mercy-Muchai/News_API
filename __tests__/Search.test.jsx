import React from 'react';
import { shallow, wrapper, mount } from 'enzyme';
import shallowToJSON from 'enzyme-to-json';
import Search from '../src/js/components/Search.jsx';
import App from '../src/js/components/app.jsx';

describe('Test Search component', () => {
  const wrapper = shallow(<Search />);
  it('should have a div', () => {
    expect(wrapper.node.type).toBe('div');
  });
  it('expect to find the class', () => {
    expect(wrapper.node.props.className).toBe('col-sm-12');
  });
  it('Should have a search button', () => {
    const homes = shallow(<Search />);
    const tree = shallowToJSON();
    expect(tree).toMatchSnapshot();
  });

  it("trigger Search", () => {
    const onSearch = jest.fn();
  });
});
describe ('Search button', ()=> {
 it('should render a search button', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.containsMatchingElement(<Search />)).toBe(true);
  });
});

const mockSources = {
  sources: [
    {
      id: "al-jazeera-english",
      name: "Al Jazeera English",
      description: "News, analysis from the Middle East and worldwide, multimedia and interactives, opinions, documentaries, podcasts, long reads and broadcast schedule.",
      url: "http://www.aljazeera.com",
      category: "general",
      language: "en",
      country: "us",
      urlsToLogos: {
        small: "",
        medium: "",
        large: ""
      },
      sortBysAvailable: [
        "top",
        "latest"
      ]
    },
    {
      id: "ars-technica",
      name: "Ars Technica",
      description: "The PC enthusiast's resource. Power users and the tools they love, without computing religion.",
      url: "http://arstechnica.com",
      category: "technology",
      language: "en",
      country: "us",
      urlsToLogos: {
        small: "",
        medium: "",
        large: ""
      },
      sortBysAvailable: [
        "top",
        "latest"
      ]
    },]
}
describe('Sources', () => {
  // user should search any source
  it('the array should have atleast one source', () => {
    const wrapper = mount(
      <Search />,
    );
    wrapper.setState(mockSources)
    wrapper.update();
    const sources = wrapper.find('sources-container col-sm-3')
    expect(sources.length).toBeLessThan(2);
  });
});