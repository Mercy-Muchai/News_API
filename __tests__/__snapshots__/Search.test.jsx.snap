// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Search component Should have a search button 1`] = `undefined`;
